libs/gl-client-py/src/credentials.rs

```diff
@@ -55,6 +57,25 @@ where
             UnifiedCredentials::Device(d) => d.tls_config(),
         }
     }
+
+    fn with_identity<V>(device_cert: V, device_key: V) -> Self
```

Comment:
I made `with_identity` determine which type of credentials to return based on the presence of a node_id in the cert. This signature had to be made this way, where a specific value for the UnifiedCredentials could not be explicitly defined by the user, because the UnifiedCredentials is a TlsConfigProvider and must implement the existing trait's associated function with no value-hint in the arguments.